packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_NullspaceFactory_def.hpp

```diff
@@ -178,9 +261,48 @@ namespace MueLu {
             nsValues[j*numPDEs + i] = 1.0;
           }
         }
+        if ( (int) nullspaceDim > numPDEs ) {
+           /* xy rotation */
+           ArrayRCP<Scalar> nsValues = part->getDataNonConst(numPDEs);
+           int numBlocks = nsValues.size() / numPDEs;
+           TEUCHOS_TEST_FOR_EXCEPTION(numBlocks != (int) xvals.size(), Exceptions::RuntimeError, "MueLu::NullspaceFactory::fillNullspaceVector(): number of coordinates does not match  ndofs/numPDEs.");
+           for (int j=0; j< numBlocks; ++j) {
+             nsValues[j*numPDEs + 0] = -(yvals[j]-cy);
+             nsValues[j*numPDEs + 1] =  (xvals[j]-cx);
+           }
+         }
```

Comment:
That looks like a duplication from the function `Build()`. What's the difference between this code and the one above? Can this duplicate be avoided by moving this to a private function, that is then called by both `Build()` and `fillNullspaceVector()`.

This comment also applies to the other two rotations.